Src/ElasticScale.Client/ShardManagement/Cache/PerfCounterInstance.cs

```diff
@@ -253,22 +255,25 @@ internal static bool HasCreatePerformanceCounterPermissions()
         /// </summary>
         public void Dispose()
         {
-            // If performance counter instance exists, remove it here.
-            if (_initialized)
+            lock (_lockObject)
```

Comment:
If you're going to use lock, use double-checked lock pattern to minimize lock contention

    if (initialized)
    {
      lock (lockobject)
      {
        if (initialized)
        {
           ....